src/proto/Cosmos.proto

```diff
@@ -85,6 +85,53 @@ message Message {
         string type_prefix = 3;
     }
 
+    // wasm/MsgExecuteContract
+    message WasmExecuteContract {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        // base64-encoded payload of contract execution message (refer to below)
+        string execute_msg = 3;
+
+        // used in case you are sending native tokens along with this message
+        repeated string coin = 4;
+    }
+
+    // transfer within wasm/MsgExecuteContract
+    message WasmExecuteContractTransfer {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        int64 amount = 3;
+
+        string recipient_address = 4;
+    }
+
+    // send within wasm/MsgExecuteContract
+    message WasmExecuteContractSend {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        int64 amount = 3;
+
+        string recipient_contract_address = 4;
+
+        // execute_msg to be executed in the context of recipient contract
+        string msg = 5;
+
+        // used in case you are sending native tokens along with this message
+        repeated string coin = 6;
```

Comment:
what is the `coin` used for?

Reply:
// used in case you are sending native tokens along with this message
Not a use case for our wallet transfer